src/transport/SessionHolder.h

```diff
@@ -47,6 +47,7 @@ class SessionHolder : public SessionDelegate, public IntrusiveListNodeBase
         return mSession.HasValue() && &mSession.Value().Get() == &session.mSession.Get();
     }
 
+    void GrabPairing(const SessionHandle & session); // Should be only used inside CASE/PASE pairing.
```

Comment:
I can't comment on the line below, so doing it here.

Why does `SessionHolder` have a reference counted handle to a session? It's effectively holding a weak reference to a session, where have a SessionHolder doesn't actually alter the lifetime of the underlying Session object.

We know this to be true due to the existence of the `NotifySessionReleased` method on `Session`, which effectively ignores any reference counts the holders may have added to the session.

I'd think this is more the correct thing:

```
Transport::Session *mSession = nullptr;
```

Reply:
The statement is true for `SecureSession,` but not for `UnauthenticatedSession`. The counter in `SessionHolder` serves eviction logic for `UnauthenticatedSession`

Reply:
Updated as per long Slack thread to now shift to this new scheme for a weak_ptr like scheme.